### ISSUE TITLE
kubectl rollout history deployment -o yaml --revision N seems to ignore the revision parameter and always use the most recent one.

### ISSUE THREAD

**Issue by MarkKharitonov** [kubernetes/kubernetes#112935]
### What happened?

```
student@master:~$ k create deployment httpd --image httpd
deployment.apps/httpd created
student@master:~$ k get pod -l app=httpd
NAME                     READY   STATUS    RESTARTS   AGE
httpd-676d9bc46d-tk94g   1/1     Running   0          74s
student@master:~$ k set image deployment.apps/httpd httpd=BOGUS_IMAGE
deployment.apps/httpd image updated
student@master:~$ k get pod -l app=httpd
NAME                     READY   STATUS             RESTARTS   AGE
httpd-676d9bc46d-dldrf   1/1     Running            0          63s
httpd-9fc9cc7c8-8bdfj    0/1     InvalidImageName   0          6s
student@master:~$ k rollout history deployment httpd
deployment.apps/httpd
REVISION  CHANGE-CAUSE
1         <none>
2         <none>

student@master:~$ k rollout history deployment httpd --revision 2 | grep -iE "revision\b"
deployment.apps/httpd with revision #2
student@master:~$ k rollout history deployment httpd --revision 1 | grep -iE "revision\b"
deployment.apps/httpd with revision #1
student@master:~$ k rollout history deployment httpd --revision 2 -o yaml | grep -iE "revision\b"
    deployment.kubernetes.io/revision: "2"
student@master:~$ k rollout history deployment httpd --revision 1 -o yaml | grep -iE "revision\b"
    deployment.kubernetes.io/revision: "2"
student@master:~$
```

### What did you expect to happen?

```
student@master:~$ k rollout history deployment httpd --revision 1 -o yaml | grep -iE "revision\b"
    deployment.kubernetes.io/revision: "1"
student@master:~$
```

### How can we reproduce it (as minimally and precisely as possible)?

See the **What happened?** section.

### Kubernetes version

<details>

```console
$ kubectl version
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.24.1
Kustomize Version: v4.5.4
Server Version: v1.24.5
```

</details>

### Cloud provider

<details>
GCE VMs, unmanaged K8s cluster created in the context of the LFD259 course (CKAD Linux Foundation)
</details>

### OS version

<details>

```console
student@master:~$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.5 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.5 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
student@master:~$ uname -a
Linux master 5.15.0-1018-gcp #24~20.04.1-Ubuntu SMP Mon Sep 12 06:14:01 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
student@master:~$
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by r3motecontrol** [comment]
It appears this was fixed in PR [111093](https://github.com/kubernetes/kubernetes/pull/111093)

**Comment by MarkKharitonov** [comment]
Oh, great. 
What k8s version can we expect to have this fix?
Thank you.

**Comment by pacoxu** [comment]
> What k8s version can we expect to have this fix?

This will be fixed in the next release v1.26. It is not cherry-picked to v1.25 or before yet. Cherry-pick may be preferred.

**Comment by MarkKharitonov** [comment]
Got it. Thank you.
